app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/agenda/AgendaViewModel.kt

```diff
@@ -34,6 +35,7 @@ class AgendaViewModel(
     private val taskRepository: TaskRepository,
     private val reminderRepository: ReminderRepository,
     private val eventRepository: EventRepository,
+    private val scheduleSyncAgendaSchedulerImpl: ScheduleSyncAgendaSchedulerImpl
```

Comment:
Never depend on the implementation directly. You not only break the layering here, since you access a data dependency in presentation, but this also defeats the whole purpose of why we have abstractions in the first place (to swap them out with different implementations)